tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,171 @@
+import copy
+import pytest
+import re
+
+from configuration import available_ports, TLS13_CIPHERS, ALL_TEST_CURVES, MINIMAL_TEST_CERTS
+from common import ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process  # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version, to_bytes
+
+PADDING_SIZE_SMALL = 250
+PADDING_SIZE_MEDIUM = 1000
+PADDING_SIZE_MAX = 1 << 14
+
+PADDING_SIZES = [
+    PADDING_SIZE_SMALL,
+    PADDING_SIZE_MEDIUM,
+    PADDING_SIZE_MAX
+]
+
+# arbitrarily large payload size
+PAYLOAD_SIZE = 1024
+
+# regex that matches server/client 'write to' with the TLS header
+OPENSSL_WRITE_TO_REGEX = r"write to [\w >\(\)\[\]=]*\\n0000 - 17 03 03 [0-9A-Fa-f]{2} [0-9A-Fa-f]{2}"
+
```

Comment:
The danger of regexes is always that they become unreadable and then become wrong. Can this be simplified? Which parts are absolutely required? Do you have an example of the output you're trying to match?

Reply:
We need to match an output that is an openssl write + the size of the record, `write to 0x272edf0 [0x274a803] (1271 bytes => 1271 (0x4F7))
0000 - 17 03 03 04 f2` will be matched. From there, we can verify if the record size `0x04f2` is expected.
```
write to 0x272edf0 [0x274a803] (1271 bytes => 1271 (0x4F7))
0000 - 17 03 03 04 f2 fc 2c 43-f7 41 05 96 f1 4b eb 5a   ......,C.A...K.Z
```

We could also match the `1271 bytes` part.

Reply:
You'd have to consider the header length to use `1271 bytes`, but that does seem reasonable. You can also use more non-greedy wildcards, since we don't care about the `0x272edf0 [0x274a803]`.

If you're using the `1271 bytes`, you could also first grab all lines that start with 'write to". Then you'd just need to match `[0-9]+ bytes` in those lines, which is a much more readable regex.

Reply:
Oh, and you should probably be using capture groups

Reply:
If we're just checking application records then we would necessarily need to match `17 03 03` as well? I can switch to capture groups but I don't know if you would consider that simplifying the regex

Reply:
Correct, that's why it's unfortunate that [you need to care about the record type](https://github.com/aws/s2n-tls/pull/3715#discussion_r1066156776). And yes, just switching to capture groups won't simplify the regex, you'll need to remove any unnecessary parts and maybe reconsider your algorithm. Maybe it would be more readable if you used multiple simpler regexes to get what you need-- for example, first match "write to" + the next line, then match the record header length in that? You may need to experiment a little.